### PR TITLE
feat - completed wishlist functionality

### PR DESCRIPTION
**Pull request by alcatraz69**

### CODE REVIEW

**Comment by sandeepmehta2155** [comment]
Considering UI routing isn't smooth

**Comment by sandeepmehta2155** [comment]
Instead of Switch Route make use of Router Route and upgrade react-router-dom to beta 6 ver, I think routing is getting delayed due to this issue only

**Comment by sandeepmehta2155** [comment]
Try to make code cleaner by inheriting from index.js avoid "./Components/NavBar/navbar in app.js

**Comment by sandeepmehta2155** [comment]
Can use backEnd for importing data : initialData

**Comment by sandeepmehta2155** [comment]
Can create faker from backend too

**Comment by sandeepmehta2155** [comment]
Refer this for rendering your images : https://gtmetrix.com/properly-size-images.html

**Comment by alcatraz69** [comment]
> Instead of Switch Route make use of Router Route and upgrade react-router-dom to beta 6 ver, I think routing is getting delayed due to this issue only

Had some issues with version 6 so used the current version. But will work on the routing thanks!

**Comment by alcatraz69** [comment]
> Can use backEnd for importing data : initialData

yeah working on it!

**Comment by alcatraz69** [comment]
> Refer this for rendering your images : https://gtmetrix.com/properly-size-images.html

Thanks for the resource, will implement this!

**Comment by alcatraz69** [comment]
> Try to make code cleaner by inheriting from index.js avoid "./Components/NavBar/navbar in app.js

I have made the changes. Thanks for your suggestion.

File: src/Components/FilterCard/FilterCard.js [hunk 1]
```diff
@@ -32,13 +34,15 @@ export default function FilterCard(){
           <legend>Preferences</legend>
           <label>
             <input
+            className={styles.radioBtn}
               type="checkbox"
               onClick={() => dispatch({ type: "OUTOFSTOCK" })}
             />
             inclue out of stock products
```
Comment (themohammadsa) [inline src/Components/FilterCard/FilterCard.js]: a typo error here. "include"

File: src/Components/FilterCard/FilterCard.js [hunk 2]
```diff
@@ -32,13 +34,15 @@ export default function FilterCard(){
...
           </label>
           <label>
             <input
+            className={styles.radioBtn}
               type="checkbox"
               onClick={() => dispatch({ type: "WITHFASTDELIVERY" })}
```
Comment (themohammadsa) [inline src/Components/FilterCard/FilterCard.js]: it's better to use 'underscore' between two words. eg: "WITH_FAST_DELIVERY"

File: src/store/ProductContext.js [hunk 3]
```diff
@@ -59,7 +59,41 @@ export const ProductsContextProvider=({children})=>{
...
+                      return{
+                          ...state,
+                          cartItems:state.cartItems.map(product=>product.id===action.payload?{...product,quantity:product.quantity>1?product.quantity-1:product.quantity}:product)
+                      };
+            
+            case "ADD_TO_CART":
+      
+                   return{
+                      ...state,
+                        cartItems:[...state.cartItems,...state.products.filter(product=>product.id===action.payload).map(item=>({...item,inCart:true,quantity:1}))],
```
Comment (themohammadsa) [inline src/store/ProductContext.js]: could you give appropriate spacing between the keys, and spacing between the object's values.

**Review by themohammadsa — CHANGES_REQUESTED** [review]
Hello Premsai!

Your app looks great! The code is nice. I have a few suggestions based on my observations:

1. The footer on the cart page does not remain in the bottom of the page.
2. The slider for price label is not working.
3. The filter and sort options are not available on the mobile version.

**Comment by alcatraz69** [comment]
> Hello Premsai!
> 
> Your app looks great! The code is nice. I have a few suggestions based on my observations:
> 
> 1. The footer on the cart page does not remain in the bottom of the page.
> 2. The slider for price label is not working.
> 3. The filter and sort options are not available on the mobile version.

Hey Mohammad! I have made all the changes, please have a look at it and approve. Thanks for your review.

File: src/Components/FilterCard/FilterCard.js (hunk 1 above)
Comment (alcatraz69) [reply src/Components/FilterCard/FilterCard.js]: Thanks for pointing it out I Have changed it

File: src/Components/FilterCard/FilterCard.js (hunk 2 above)
Comment (alcatraz69) [reply src/Components/FilterCard/FilterCard.js]: done

**Review by themohammadsa — APPROVED** [review]
Hello Premsai! Good job.

Everything looks good from my end. Go ahead with the merge from my end.

**Review by sandeepmehta2155 — APPROVED** [review]
Test

**Pull request merged**